core/project.go

```diff
@@ -308,7 +308,7 @@ func (p *Project) getResolvers(ctx context.Context, gw bkgw.Client) (router.Reso
 	return resolvers, nil
 }
 
-func (p *Project) getResolver(ctx context.Context, gw bkgw.Client) (graphql.FieldResolveFn, error) {
+func (p *Project) getResolver(ctx context.Context, gw bkgw.Client, progSock *Socket) (graphql.FieldResolveFn, error) {
```

Comment:
nit: not a blocker, but the same problem you solved here applies to any dagger exec where `ExperimentalPrivilegedNesting` is true, so it would be good to make the same change in `core/container.go` too someday.

(Honestly, this project code should probably be refactored to just call that container code so we can get rid of the duplication, but I can take care of that separately. There's some other refactorization we need here anyways to enable sdk-specific cache mounts.)

Reply:
Done! Had to pass the socket through a few places because of the default-exec behavior, but nothing too crazy.